### ISSUE TITLE
Allow classes' dynamic attribute assignments outside of class definition

### ISSUE THREAD

**Issue by cheny0y0** [microsoft/pyright#6611]
**Dynamic attribute assignment through `.`**
```python
class Foo:
    """test method will implement later"""
    pass
if __debug__:
    def test(self, x):
        assert x, x
        return x
else:
    def test(self, x):
        if not x:
            raise AssertionError(x)
        return x
Foo.test = test
```
It reports `reportGeneralTypeIssues` on the first `test` on the last line.

**Describe the solution you’d like**
Do not report this as an error, just like replace `Foo.test = test` with `setattr(Foo, "test", test)` which does not report an error.  
By the way, is this by-design?
